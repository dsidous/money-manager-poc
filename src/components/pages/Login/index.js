import React, { PureComponent } from 'react';
import { View, Image, TouchableOpacity, KeyboardAvoidingView } from 'react-native';
import { connect } from 'react-redux';
import { StyledTextInput, StyledText, StyledButton, StyledView } from '../../atoms';
import { FETCH_USER_DATA } from "../../../store/constants";
import PropTypes from 'prop-types';

import usernameImg from '../../../assets/images/userlogin/username.png';
import passwordImg from '../../../assets/images/userlogin/password.png';
import eyeImg from '../../../assets/images/userlogin/eye_black.png';

const styles = {
  container: {
    flex: 1,
    justifyContent: 'center',
    paddingHorizontal: 25,
  },
  firstContainer: {
    flexDirection: 'column',
    justifyContent: 'center',
  },
  centered: {
    textAlign: 'center'
  },
  small: {
    fontSize: 12
  },
  inputWrapper: {
    alignItems: 'center',
    marginBottom: 10,
  },
  inlineImg: {
    position: 'absolute',
    zIndex: 99,
    width: 22,
    height: 22,
    left: 30,
    top: 9,
    tintColor: 'rgba(0,0,0,0.2)',
  },
  btnEye: {
    position: 'absolute',
    top: 10,
    right: 28,
  },
  iconEye: {
    width: 25,
    height: 25,
    tintColor: 'rgba(0,0,0,0.2)',
  },
  button: { marginTop: 15 },
  logoImgStyle: {
    width: '90%',
    height: '28%',
    margin: '5%',
    borderRadius: 8,
  }
};


class Login extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      userName: '',
      password: '',
      loginDisable: true,
      showPass: true,
      press: false,
      isError: false,
      errorMessage: 'User name and/or password is invalid!',
    };
    this.showPass = this.showPass.bind(this);
  }

  componentDidMount() {
    this.props.dispatch({ type: FETCH_USER_DATA })
  }

  onChangeText = (key, val) => {
    this.setState({
      [key]: val
    })
  }

  loginAuthentication = (e) => {

    const result1 = this.props.userList;
    const result2 = {
      userName: this.state.userName,
      password: this.state.password
    };

    let result = result1.filter(
      obj => result2.userName === obj.userName
        && result2.password === obj.password
    );

    if (result.length > 0) {
      this.props.navigation.navigate('Setup');
      this.setState({
        isError: false
      });
    }
    else {
      this.setState({
        isError: true
      });
    }
    e.preventDefault();
  }
  showPass() {
    this.state.press === false
      ? this.setState({ showPass: false, press: true })
      : this.setState({ showPass: true, press: false });
  }

  render() {
    const { userName, password } = this.state;
    const isEnabled =
      (userName.length > 0 &&
        password.length > 0);

    return (
      <KeyboardAvoidingView style={styles.container} behavior="padding">
        <StyledView style={styles.firstContainer}>
          <View style={styles.inputWrapper}>
            <Image source={usernameImg} style={styles.inlineImg} />
            <StyledTextInput
              value={this.state.userName}
              placeholder="Username"
              autoCapitalize="none"
              onChangeText={val => this.onChangeText('userName', val)}
            />
          </View>
          <View style={styles.inputWrapper}>
            <Image source={passwordImg} style={styles.inlineImg} />
            <StyledTextInput password
              value={this.state.password}
              placeholder='Password'
              secureTextEntry={this.state.showPass}
              returnKeyType={'done'}
              autoCorrect={false}
              onChangeText={val => this.onChangeText('password', val)}
            />

            <TouchableOpacity
              activeOpacity={0.7}
              style={styles.btnEye}
              onPress={this.showPass}>
              <Image source={eyeImg} style={styles.iconEye} />
            </TouchableOpacity>
          </View>
          <View>
            {
              this.state.isError ?
                <StyledText size='small' color style={styles.centered}>{this.state.errorMessage}</StyledText>
                : <StyledText />
            }
          </View>
          <View style={{ paddingHorizontal: 80 }}>
            <StyledButton
              style={styles.button} disabled={!isEnabled}
              onPress={(e) => this.loginAuthentication(e)}
            >
              LOGIN
            </StyledButton>
          </View>
        </StyledView>
      </KeyboardAvoidingView>
    );
  }
}

const mapStateToProps = (state) => ({
  userList: state.login.userList,
  selected: state.login.selected
})

Login.propTypes = {
  userList: PropTypes.array,
  navigation: PropTypes.object,
}

export default connect(mapStateToProps)(Login);
