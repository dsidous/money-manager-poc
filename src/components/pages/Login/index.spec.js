import React from 'react';
import { shallow, configure } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import Login from './index';

configure({ adapter: new Adapter() });

describe('login', () => {

  const initialState = {
    login: {
      userList: [
        {
          user: 'user1',
          userName: 'username1',
          emailID: 'email1',
          password: 'password1',
        },
        {
          user: 'user2',
          userName: 'username2',
          emailID: 'email2',
          password: 'password2',
        },
      ],
    },
  };

  const mockStore = configureStore();
  const store = mockStore(initialState);
  const render = shallow(<Login store={store} />);

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });
});
