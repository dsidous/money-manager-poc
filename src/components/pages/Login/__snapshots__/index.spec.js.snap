// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Login)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "selected": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "userList": Array [
        Object {
          "emailID": "email1",
          "password": "password1",
          "user": "user1",
          "userName": "username1",
        },
        Object {
          "emailID": "email2",
          "password": "password2",
          "user": "user2",
          "userName": "username2",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "selected": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "userList": Array [
          Object {
            "emailID": "email1",
            "password": "password1",
            "user": "user1",
            "userName": "username1",
          },
          Object {
            "emailID": "email2",
            "password": "password2",
            "user": "user2",
            "userName": "username2",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
